encoding/rs/icicle/extend_poly.go

```diff
@@ -0,0 +1,40 @@
+//go:build icicle
+
+package icicle
+
+import (
+	"sync"
+
+	"github.com/Layr-Labs/eigenda/encoding/icicle"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/core"
+	icicle_bn254 "github.com/ingonyama-zk/icicle/v3/wrappers/golang/curves/bn254"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/curves/bn254/ntt"
+	"github.com/ingonyama-zk/icicle/v3/wrappers/golang/runtime"
+)
+
+type RsIcicleComputeDevice struct {
+	NttCfg core.NTTConfig[[icicle_bn254.SCALAR_LIMBS]uint32]
+	Device runtime.Device
+}
+
+// Encoding Reed Solomon using FFT
+func (g *RsIcicleComputeDevice) ExtendPolyEval(coeffs []fr.Element) ([]fr.Element, error) {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	var evals []fr.Element
+	g.NttCfg.BatchSize = int32(1)
+	runtime.RunOnDevice(&g.Device, func(args ...any) {
```

Comment:
If the GPU is enabled, then this func will be executed on GPU by the runtime?

Reply:
yes because the device variable will be set to the GPU device here https://github.com/Layr-Labs/eigenda/blob/dfe247d701aea8685f265d34b2f553bfec8863ff/encoding/icicle/device_setup.go#L88

Reply:
For performance, it looks the main data movement to/from GPU will be the coefficients and the resulting evaluations?